File: other_langs/sqsSample.java
/*
 * Copyright 2010-2013 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *  http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
package com.amazonaws.samples;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.Writer;

import com.amazonaws.AmazonClientException;
import com.amazonaws.AmazonServiceException;
import com.amazonaws.regions.Region;
import com.amazonaws.regions.Regions;
import com.amazonaws.services.sqs.AmazonSQS;
import com.amazonaws.services.sqs.AmazonSQSClient;

public class S3Sample {

    public static void main(String[] args) throws IOException {
        AmazonSQS sqs = new AmazonSQSClient();
        Region usWest2 = Region.getRegion(Regions.US_WEST_2);
        sqs.setRegion(usWest2);
        sqs.setEndpoint("http://localhost:5000");

        String queueName = "my-first-queue";
        sqs.createQueue(queueName);

        System.out.println("Listing queues");
        for (String queue_url: sqs.listQueues().getQueueUrls()) {
            System.out.println(" - " + queue_url);
        }
        System.out.println();

    }

}

File: other_langs/sqsSample.scala
package com.amazonaws.examples

import com.amazonaws.client.builder.AwsClientBuilder
import com.amazonaws.regions.{Region, Regions}
import com.amazonaws.services.sqs.AmazonSQSClientBuilder

import scala.jdk.CollectionConverters._

object QueueTest extends App {
  val region = Region.getRegion(Regions.US_WEST_2).getName
  val serviceEndpoint = "http://localhost:5000"

  val amazonSqs =  AmazonSQSClientBuilder.standard()
    .withEndpointConfiguration(
      new AwsClientBuilder.EndpointConfiguration(serviceEndpoint, region))
    .build

  val queueName = "my-first-queue"
  amazonSqs.createQueue(queueName)

  val urls = amazonSqs.listQueues().getQueueUrls.asScala
  println("Listing queues")
  println(urls.map(url => s" - $url").mkString(System.lineSeparator))
  println()
}
